samtranslator/model/sam_resources.py

```diff
@@ -2194,3 +2202,108 @@ def _construct_appsync_schema(self, api_id: Intrinsicable[str]) -> GraphQLSchema
         schema.DefinitionS3Location = self.SchemaUri
 
         return schema
+
+
+class SamGraphQLDataSource(SamResourceMacro):
+    """SAM GraphQL Data Source Macro (WIP)."""
+
+    # TODO: This currently only supports DataSource with a defined ServiceRoleArn.
+    #       The proposed connector change will come in a follow up PR.
+    resource_type = "AWS::Serverless::GraphQLDataSource"
+    property_types = {
+        "ApiId": Property(True, one_of(IS_STR, IS_DICT)),
+        "Name": Property(False, IS_STR),
+        "Type": Property(True, IS_STR),
+        "Description": PassThroughProperty(False),
+        "ServiceRoleArn": Property(False, IS_STR),
+        "Permissions": Property(False, list_of(IS_STR)),
+        "DataSourceConfig": Property(True, IS_DICT),
+    }
+
+    ApiId: Intrinsicable[str]
```

Comment:
Think we don't resolve intrinsics anymore since we recommend adding [AWS::LanguageExtensions](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-languageextension-transform.html) to the template so this shouldn't be `Intrinsicable`